cheroot/server.py

```diff
@@ -341,16 +342,53 @@ def __next__(self):
     next = __next__
 
 
-class KnownLengthRFile:
+class RFile:
+    def __init__(self, rfile, wfile, conn):
+        self.rfile = rfile
+        self.wfile = wfile
+        self.conn = conn
+
+    def _send_100_if_needed(self):
+        if self.conn.they_are_waiting_for_100_continue:
+            mini_headers = ()
+            go_ahead = h11.InformationalResponse(status_code=100, headers=mini_headers)
+            bytes_out = self.conn.send(go_ahead)
+            self.wfile.write(bytes_out)
```

Comment:
let's use a guard check style here
```suggestion
        if not self.conn.they_are_waiting_for_100_continue:
            return
        mini_headers = ()
        go_ahead = h11.InformationalResponse(status_code=100, headers=mini_headers)
        bytes_out = self.conn.send(go_ahead)
        self.wfile.write(bytes_out)
```